device/core/src/sak_signature_provider.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import {
+  SignatureProvider,
+  SharedAccessSignature,
+  errors
+} from 'azure-iot-common';
+
+export class SharedAccessKeySignatureProvider implements SignatureProvider {
+  private _tokenValidTimeInSeconds: number = 3600; // 1 hour
+
+  constructor(
+    private _sharedAccessKey: string,
+    tokenValidTimeInSeconds?: number
+  ) {
+    // Codes_SRS_NODE_SAK_SIG_PROVIDER_13_003: [ The constructor shall throw an ArgumentError if the _sharedAccessKey parameter is falsy. ]
+    if (!this._sharedAccessKey) {
+      throw new errors.ArgumentError(
+        '_sharedAccessKey cannot be "' + this._sharedAccessKey + '".'
+      );
+    }
+
+    // Codes_SRS_NODE_SAK_SIG_PROVIDER_13_004: [ The constructor shall save the tokenValidTimeInSeconds parameter if supplied. If not, it shall default to 3600 seconds (1 hour). ]
+    if (tokenValidTimeInSeconds)
+      this._tokenValidTimeInSeconds = tokenValidTimeInSeconds;
```

Comment:
surprised the linter is not barking at you for skipping curly braces here.

Reply:
Ah. I'll add the braces. Yeah, linter didn't say anything.